pkg/core/workflow_store.go

```diff
@@ -2,43 +2,104 @@ package core
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/fuseml/fuseml-core/gen/workflow"
+	"github.com/fuseml/fuseml-core/pkg/domain"
 )
 
+// storableWorkflow holds the workflow and the codesets assigned to it
+type storableWorkflow struct {
+	// Workflow assigned to the codeset
+	workflow *workflow.Workflow
+	// Codesets assigned to the workflow
+	assignedCodesets []*domain.Codeset
+}
+
 // WorkflowStore describes in memory store for workflows
 type WorkflowStore struct {
-	items map[string]*workflow.Workflow
+	items map[string]*storableWorkflow
 }
 
 // NewWorkflowStore returns an in-memory workflow store instance
 func NewWorkflowStore() *WorkflowStore {
 	return &WorkflowStore{
-		items: make(map[string]*workflow.Workflow),
+		items: make(map[string]*storableWorkflow),
 	}
 }
 
-// Find returns a workflow identified by id
-func (ws *WorkflowStore) Find(ctx context.Context, name string) *workflow.Workflow {
-	return ws.items[name]
+// GetWorkflow returns a workflow identified by its name
+func (ws *WorkflowStore) GetWorkflow(ctx context.Context, name string) *workflow.Workflow {
+	if _, exists := ws.items[name]; exists {
+		return ws.items[name].workflow
+	}
+	return nil
 }
 
-// GetAll returns all workflows that matches a given name.
-func (ws *WorkflowStore) GetAll(ctx context.Context, name string) (result []*workflow.Workflow) {
+// GetAllWorkflows returns all workflows or a wothat matches a given name.
+func (ws *WorkflowStore) GetAllWorkflows(ctx context.Context, name *string) (result []*workflow.Workflow) {
 	result = make([]*workflow.Workflow, 0, len(ws.items))
-	for _, w := range ws.items {
-		if name == "all" || w.Name == name {
-			result = append(result, w)
-		}
+	if name != nil {
```

Comment:
Thank you for getting rid of the "all" name wildcard, I didn't particularly like it.